rowboat/plugins/msghandler.py

```diff
@@ -0,0 +1,367 @@
+                overrides.update(obj)
+
+            for obj in config.commands.get_lockdown_rules(tag.name):
+                overrides.update(obj)
+
+            override_chans = overrides.get('out')
+
+            if hasattr(override_chans, 'get'):
+                lockchans = override_chans.get('channels', lockchans)
+                locktype = next(iter(overrides))
+
+            allowtg = False
+            if allowroles is not None:
+                member = guild.get_member(author)
+                if member:
+                    for rid in member.roles:
+                        if rid in allowroles:
+                            allowtg = True
+
+            bypasstg = True if global_bypass or allowtg else False
+
+            if (not bypasstg and user_level < level):
+                try:
+                    event.delete()
+                    if not config.commands.dm_denied:
+                        author.user.open_dm().send_message('You do not have enough permissions to run that command')
+                    return
+                except:
+                    return
+                return
+
+            if bypasstg or event.channel.id in lockchans or len(lockchans) == 0:
+                pass
+            else:
+                try:
+                    event.delete()
+                    if not config.commands.dm_denied:
+                        author.user.open_dm().send_message('That cannot be run in the channel **{}**'.format(channel_name))
+                    return
+                except:
+                    return
+                return
+
+            Tag.update(times_used=Tag.times_used + 1).where(
+                (Tag.guild_id == tag.guild_id) &
+                (Tag.name == tag.name)
+            ).execute()
+
+            urls = URL_RE.findall(tag.content)
+            if len(urls) > 0 and len(tag.content.split()) == 1 and not (tag.content.startswith("<") and tag.content.endswith(">")):
+                for url in urls:
+                   r = requests.get(tag.content)
+                try:
+                    r.raise_for_status()
+                    ext = 'png'
+                    if (tag.content.endswith('.gif')):
+                        ext='gif'
+                    elif (tag.content.endswith('.jpg')):
+                        ext='jpg'
+                    elif (not tag.content.endswith('.gif')):
+                        channel.send_message(u'{}'.format(
+                            tag.content
+                        ))
+                        return
+                    channel.send_message('', attachments=[('attach.{}'.format(ext), r.content)])
+                    return
+                except:
+                    return
+
+            channel.send_message(u'{}'.format(
+                tag.content
+            ))
+            return
+        return
+
+    @Plugin.listen('MessageCreate')
+    def on_message_create(self, event):
```

Comment:
would it matter if we renamed it `on_dm_message_create` so it's not confusing?

Reply:
fixed